app-ui/src/pages/traces/GuardrailSuggestions.tsx

```diff
@@ -0,0 +1,460 @@
+  const [showPolicySynthesisModal, setShowPolicySynthesisModal] =
+    React.useState(false);
+
+  // refreshes the list of stored suggested guardrails
+  const refreshStoredPolicies = async () => {
+    try {
+      // Use query parameters to filter policies on the server side
+      const minDetectionRate = 0.7; // Filter out policies with detection rate below 70%
+      const successOnly = true; // Only include successful policies
+
+      const storedPoliciesResponse = await fetch(
+        `/api/v1/dataset/byid/${dataset.id}/generated-policies?min_detection_rate=${minDetectionRate}&success_only=${successOnly}`
+      );
+
+      if (!storedPoliciesResponse.ok) {
+        console.error(
+          "Failed to fetch stored policies:",
+          storedPoliciesResponse.status
+        );
+        return;
+      }
+
+      const storedPoliciesData = await storedPoliciesResponse.json();
+      setStoredPolicies(storedPoliciesData.policies || []);
+    } catch (error) {
+      console.error("Error fetching stored policies:", error);
+    }
+  };
+
+  // refreshes the list of active guardrail suggestion jobs
+  const refreshActiveJobs = async () => {
+    try {
+      const jobsResponse = await fetch(
+        `/api/v1/dataset/byid/${dataset.id}/jobs`
+      );
+      if (jobsResponse.ok) {
+        const data = await jobsResponse.json();
+        // Filter for policy synthesis jobs
+        let synthJobs = data.filter(
+          (job: any) => job.extra_metadata.type === "policy_synthesis"
+        ) as PolicyJob[];
+
+        // filter down to jobs that are not already done
+        synthJobs = synthJobs.filter((job) =>
+          [JOB_STATUS.PENDING, JOB_STATUS.RUNNING].includes(
+            job.extra_metadata.status
+          )
+        );
+
+        setPolicyJobs(synthJobs);
+      }
+    } catch (error) {
+      console.error("Error fetching active jobs:", error);
+    }
+  };
+
+  // on mount, refresh
+  React.useEffect(() => {
+    // fetch active jobs and stored policies
+    refreshStoredPolicies();
+    refreshActiveJobs();
+  }, [dataset]);
+
+  // when there are pending policy jobs, regularly refresh stored and active jobs
+  React.useEffect(() => {
+    if (policyJobs.length > 0) {
+      const refresher = () => {
+        refreshStoredPolicies();
+        refreshActiveJobs();
+      };
+
+      // set and clear interval
+      const interval = setInterval(refresher, POLL_INTERVAL);
+      return () => clearInterval(interval);
+    }
+  }, [policyJobs]);
+
+  // Update handlePolicySynthesisSuccess to include better polling management
+  const onPolicySynthesisStart = (job_info) => {
+    // clear existing stored policies and active jobs
+    setStoredPolicies([]);
+    setPolicyJobs([]);
+    // Immediately fetch active jobs and start polling
+    refreshStoredPolicies();
+    refreshActiveJobs();
+  };
+
+  // track whether 'Cancel' was clicked
+  const [cancelClicked, setCancelClicked] = React.useState(false);
```

Comment:
we track this separately, so the user doesn't cancel 10 times, if it has latency